torch/utils/data/dataloader.py

```diff
@@ -127,7 +131,7 @@ def _share_dist_seed(generator, pg):
     return _shared_seed.item()
 
 
-class DataLoader(Generic[T_co]):
+class DataLoader(Generic[SampleT_co, BatchT_co]):
```

Comment:
I am not an expert on typing in python. Does the following work with this change?
```py
dl: DataLoader[Sample] = DataLoader(dataset, ...)
```
I am trying to understand if this is going to be BC breaking or not

Reply:
I was under the impression that it should not break because for some reasons it type-checked locally but upon closer inspection it unfortunately breaks. I assumed that unspecified generic parameter are assumed to be `Any` but it isn't the case